libbeat/common/kubernetes/metadata.go

```diff
@@ -26,21 +26,27 @@ import (
 
 // MetaGenerator builds metadata objects for pods and containers
 type MetaGenerator interface {
+	// ResourceMetadata generates metadata for the given kubernetes object taking to account certain filters
+	ResourceMetadata(obj Resource) common.MapStr
+
 	// PodMetadata generates metadata for the given pod taking to account certain filters
 	PodMetadata(pod *Pod) common.MapStr
 
 	// Containermetadata generates metadata for the given container of a pod
 	ContainerMetadata(pod *Pod, container string) common.MapStr
 }
 
-type metaGenerator struct {
+// MetaGeneratorConfig settings
+type MetaGeneratorConfig struct {
 	IncludeLabels          []string `config:"include_labels"`
 	ExcludeLabels          []string `config:"exclude_labels"`
 	IncludeAnnotations     []string `config:"include_annotations"`
-	IncludePodUID          bool     `config:"include_pod_uid"`
+	IncludeUID             bool     `config:"include_uid"`
 	IncludeCreatorMetadata bool     `config:"include_creator_metadata"`
```

Comment:
Are these config settings documented?

Reply:
No, I struggle a bit with these, I wonder if that means we should just hardcode them in. They were kept just in case someone wants to disable them, but I'm not sure

Reply:
Is it currently disabled by default?

As this doesn't look so costly to collect, maybe we can just add them, and they can be removed with `drop_fields`.

Reply:
yes, so far I have left them undocumented to see how much pushback they get once released, I can leave a note here to say we want to remove these settings